src/khoj/interface/web/chat.html

```diff
@@ -2170,6 +2214,24 @@
                             </svg>
                         </button>
                         <textarea id="chat-input" class="option" oninput="onChatInput()" onkeydown=incrementalChat(event) autofocus="autofocus" placeholder="Type / to see a list of commands"></textarea>
+                        <!-- Shortcut Handler for Accessing Old Messages -->
+                        <script>
+                            document.addEventListener('keydown', function(event) {
+                                const chatInput = document.getElementById('chat-input');
+                                if (document.activeElement === chatInput) {
+                                    if (event.key === 'ArrowUp') {
```

Comment:
I think we should trigger showing the chat history only when the chat input is empty. Otherwise it'll conflict with scrolling up in a multi-line chat input

Reply:
The only problem is that once the first chat history message is loaded, the chat input won't be empty anymore.

Reply:
Oh yeah good point!

Maybe we can use a different keybinding or use buttons for this or keep the up/down arrow bound to cycling through recent messages when an item from chat history was selected (vs scrolling when text was input via keyboard)?

Also will the current behaviour work off selecting recent messages work on phones (given they use touch/scroll events instead of arrow keys)?